### PR TITLE
Doc/readme fixes + improvements

### PR DESCRIPTION
**Pull request by IanConnolly**
Bunch of README additions + fixes, bonus typo in my documentation.

Will close out #51.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -21,22 +22,57 @@ details.
 git clone --depth=1 https://github.com/rust-lang/rust.vim.git ~/.vim/bundle/rust.vim
 ```
 
-[r]: https://rust-lang.org
```
Comment (steveklabnik) [inline README.md]: doesn't this break the links?

File: README.md [hunk 2]
```diff
@@ -21,22 +22,57 @@ details.
...
 
 ## Help
 
-Further help can be found in the documentation with `:help rust`.
+Detailed help can be found in the documentation with `:help rust`.
+Helptags (`:help helptags`) need to be generated for this plugin
+in order to navigate the help. Most plugin managers will do this
+automatically, but check their documentation if that is not the case.
+
+[r]: https://rust-lang.org
```
Comment (steveklabnik) [inline README.md]: oh, they're moved here, sorry

**Comment by steveklabnik** [comment]
This looks good overall, but it says it can't be merged. Mind rebasing?

**Comment by IanConnolly** [comment]
Sure!

**Pull request merged**